web/html/src/manager/maintenance/maintenance-windows.js

```diff
@@ -0,0 +1,249 @@
+            .then(item => {
+                this.setState({
+                    selected: item,
+                    action: action
+                });
+                history.pushState(null, null, "#/" + action + "/" +
+                    (this.state.type === "schedule" ? item.scheduleId : item.calendarId));
+            }).catch(this.handleResponseError);
+    }
+
+    handleDetailsAction(row) {
+        this.getDetails(row, "details");
+    }
+
+    handleEditAction(row) {
+        this.getDetails(row, "edit");
+    }
+
+    toggleActive(schedule) {
+        Object.assign(schedule, {
+            active: !(schedule.active)
+        });
+        this.updateSchedule(schedule);
+    }
+
+    update(item) {
+        return Network.post(
+            "/rhn/manager/api/maintenance/" + this.state.type + "/save",
+            JSON.stringify(item),
+            "application/json"
+        ).promise.then((_) => {
+            const successMsg = <span>{t(
+                (this.state.type === "schedule" ? "Schedule " : "Calendar ") +
+                "successfully" + (this.state.action === "create" ? " created." : " updated."))}</span>
+            const msgs = this.state.messages.concat(MessagesUtils.info(successMsg));
+
+            while (msgs.length > messagesCounterLimit) {
+                msgs.shift();
+            }
+
+            this.setState({
+                messages: msgs
+            });
+
+            this.handleForwardAction();
+        }).catch(this.handleResponseError);
+    }
+
+    delete(item) {
+        return Network.del("/rhn/manager/api/maintenance/" + this.state.type + "/delete",
+            JSON.stringify(item),
+            "application/json")
+            .promise.then((_) => {
+                this.setState({
+                    messages: MessagesUtils.info(
+                        (this.state.type === "schedule" ? "Schedule " : "Calendar ") +
+                        "\'" + (this.state.type === "schedule" ? item.scheduleName : item.calendarName) +
+                        "\' has been deleted."
+                    )
+                });
+                this.handleForwardAction();
+            })
+            .catch(data => {
+                const errorMsg = MessagesUtils.error(
+                    t("Error when deleting the " + this.state.type));
+                let messages = (data && data.status === 400)
+                    ? errorMsg
+                    : Network.responseErrorMessage(jqXHR);
+                this.setState({
+                    messages: messages
+                });
+            });
+    }
+
+    refreshCalendar(item) {
+        return Network.post(
+            "/rhn/manager/api/maintenance/calendar/refresh",
+            JSON.stringify(item),
+            "application/json"
+        ).promise.then((_) => {
+            const successMsg = <span>{t("Calendar successfully refreshed")}</span>
+            const msgs = this.state.messages.concat(MessagesUtils.info(successMsg));
```

Comment:
So we are not receiving any data back from the backend? Nothing about the correctness of the action completed?

Anyway, I'd do this in one line only: `MessagesUtils.info(t("Calendar successfully refreshed"))`

Reply:
Yes we do, the calendar data will be updated. This message will only be displayed in case everything went right. If not we receive an error message from the backend.